com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStepTest.java

```diff
@@ -235,4 +242,140 @@ protected CreateAppStep createStep() {
+                if (simpleService.name.equals(serviceName)) {
+                    return simpleService.toCloudServiceExtended();
+                }
+            }
+            return new CloudServiceExtended(null, serviceName);
+        }
+
+        private void loadParameters() {
+            dockerInfo = createDockerInfo();
+            application = stepInput.applications.get(stepInput.applicationIndex);
+        }
+
+        private DockerInfo createDockerInfo() {
+            String image = "cloudfoundry/test-app";
+            String username = "someUser";
+            String password = "somePassword";
+            DockerInfo dockerInfo = new DockerInfo(image);
+            DockerCredentials dockerCredentials = new DockerCredentials(username, password);
+            dockerInfo.setDockerCredentials(dockerCredentials);
+
+            return dockerInfo;
+        }
+
+        private void prepareClient() {
+            Mockito.when(configuration.getPlatformType())
+                .thenReturn(stepInput.platform);
+        }
+
+        @Test
+        public void testWithDockerImageXS2() {
+            stepInput = initStepInputForTestWithDockerImageFor(PlatformType.XS2);
+            setUp();
+
+            step.execute(context);
+            assertStepFinishedSuccessfully();
+
+            validateClient();
+        }
+
+        @Test
+        public void testWithDockerImageCF() {
+            stepInput = initStepInputForTestWithDockerImageFor(PlatformType.CF);
+            setUp();
+
+            step.execute(context);
+            assertStepFinishedSuccessfully();
+
+            validateClient();
+            validateApplicationUpdateCF();
+        }
+
+        private StepInput initStepInputForTestWithDockerImageFor(PlatformType platformType) {
+            StepInput st = new StepInput();
+
+            CloudApplicationExtended cloudApplicationExtended = createFakeCloudApplicationExtended();
+
+            st.applicationIndex = 0;
+            st.platform = platformType;
+            st.applications = Arrays.asList(cloudApplicationExtended);
+
+            return st;
+        }
+
+        private CloudApplicationExtended createFakeCloudApplicationExtended() {
+            CloudApplicationExtended cloudApplicationExtended = new CloudApplicationExtended(null, "application1");
+
+            cloudApplicationExtended.setInstances(1);
+            cloudApplicationExtended.setMemory(0);
+            cloudApplicationExtended.setDiskQuota(512);
+            cloudApplicationExtended.setEnv(Collections.emptyMap());
+            cloudApplicationExtended.setServices(Collections.emptyList());
+            cloudApplicationExtended.setServiceKeysToInject(Collections.emptyList());
+            cloudApplicationExtended.setUris(Collections.emptyList());
+            cloudApplicationExtended.setDockerInfo(dockerInfo);
+
+            return cloudApplicationExtended;
+        }
+
+        private void validateApplicationUpdateCF() {
```

Comment:
Why are we validating whether the application's staging has been updated? Is this relevant to the docker scenario in some way?